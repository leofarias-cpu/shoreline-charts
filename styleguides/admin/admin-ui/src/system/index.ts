export { jsx, SxStyleProp, SxProps } from 'theme-ui'
export { keyframes, css } from '@emotion/core'
export { default as styled } from '@emotion/styled'
export { get } from '@theme-ui/css'
export * from './theme'
export { default as merge } from 'deepmerge'

// Tokens
export { typographyTokens, TypographyTokensProps } from './tokens/typography'
export { flexTokens, FlexTokensProps } from './tokens/flexbox'
export { gridTokens, GridTokensProps } from './tokens/grid'
export { layoutTokens, LayoutTokensProps } from './tokens/layout'
export { spaceTokens, SpaceTokensProps } from './tokens/space'
export { colorTokens, ColorTokensProps } from './tokens/color'
export { borderTokens, BorderTokensProps } from './tokens/border'
export { positionTokens, PositionTokensProps } from './tokens/position'
export { sxTokens, SxTokensProps } from './tokens/sx'
export { variantToken, VariantTokenProps } from './tokens/variant'
