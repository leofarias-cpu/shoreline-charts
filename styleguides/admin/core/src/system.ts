export {
  Global,
  css,
  jsxs,
  createComponent,
  isFunction,
  useTheme,
  useResponsiveValue,
  get,
  pick,
  omit,
  isObjectEmpty,
  merge,
  forwardRef,
  inlineVariant,
  StyleProp,
  StyleObject,
  ResponsiveValue,
  darken,
  lighten,
  grayscale,
  getColor,
  rotate,
  hue,
  saturation,
  lightness,
  saturate,
  desaturate,
  shade,
  tint,
  transparentize,
  alpha,
  mix,
  complement,
  invert,
  Colors,
} from '@vtex/admin-ui-system'
