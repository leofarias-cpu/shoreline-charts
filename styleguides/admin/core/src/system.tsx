export {
  useTheme,
  jsxs,
  JSXSParams,
  useResponsiveValue,
  get,
  pick,
  omit,
  isFunction,
  isObjectEmpty,
  merge,
  cleanProps,
  pickHTMLProps,
  isPropValid,
  forwardRef,
  inlineVariant,
  StyleProp,
  StyleObject,
  ResponsiveValue,
  darken,
  lighten,
  grayscale,
  getColor,
  rotate,
  hue,
  saturation,
  lightness,
  saturate,
  desaturate,
  shade,
  tint,
  transparentize,
  alpha,
  mix,
  complement,
  invert,
  Colors,
} from '@vtex/admin-ui-system'
