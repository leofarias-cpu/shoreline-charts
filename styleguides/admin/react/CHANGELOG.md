# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.2...@vtex/onda-react@0.4.3) (2021-07-20)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.1...@vtex/onda-react@0.4.2) (2021-07-19)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.0...@vtex/onda-react@0.4.1) (2021-07-16)


### Bug Fixes

* **onda-react:** remove the cleaning of options properties from jsx ([b8ec31c](https://github.com/vtex/onda/commit/b8ec31cf1e557996ea8eb7027edbcfcbe55845fb))





# [0.4.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.3.0...@vtex/onda-react@0.4.0) (2021-07-15)


### Features

* **admin-ui/ultimate-table:** support sorting ([7453750](https://github.com/vtex/onda/commit/745375001bd28bb73e8a229f14c12e78bd0385da))





# [0.3.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.2...@vtex/onda-react@0.3.0) (2021-07-15)


### Bug Fixes

* **onda-react:** jsx refs type inference ([5a24609](https://github.com/vtex/onda/commit/5a2460960c419a272f16d76a43f9575247643f2a))
* **onda-react:** jsx rendering issues ([6fbe2b5](https://github.com/vtex/onda/commit/6fbe2b585ab083a99a1aa806de3ed0e23ac2451f))
* **onda-react:** typescript inference for jsx.elements ([fb9e365](https://github.com/vtex/onda/commit/fb9e365df6a877a274bc3ea26131a4015fbbcd33))


### Features

* **onda-react:** add ComponentProps type ([e13433c](https://github.com/vtex/onda/commit/e13433c2b9611de3cd6723541fe82064bd1f10cb))





## [0.2.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.1...@vtex/onda-react@0.2.2) (2021-07-05)

**Note:** Version bump only for package @vtex/onda-react





## [0.2.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.0...@vtex/onda-react@0.2.1) (2021-07-01)


### Bug Fixes

* **onda-core:** build imports ([9d30f9e](https://github.com/vtex/onda/commit/9d30f9e743d75b03995547e4d3052aa5a510e095))





# [0.2.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.1...@vtex/onda-react@0.2.0) (2021-07-01)


### Bug Fixes

* **onda-react:** remove unnecessary state ([1465e69](https://github.com/vtex/onda/commit/1465e695a6cb9be6ba3773d0acf51194a0713ea2))
* **onda-react:** resolve jsx hof types ([2e185ac](https://github.com/vtex/onda/commit/2e185ac2dd319f30d1eb4332a13a2299a38e06cb))


### Features

* **onda-react:** rephrase createComponent as jsx ([dd157ff](https://github.com/vtex/onda/commit/dd157ffa2500ed8e911b7d2d04db65268fc7f27b))





## [0.1.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.4...@vtex/onda-react@0.1.1) (2021-06-29)


### Bug Fixes

* **onda-react:** package.json version ([fb98ac4](https://github.com/vtex/onda/commit/fb98ac411f80ae253225940a7c687cb458677e4a))





# 0.1.0-labs.4 (2021-06-24)


### Features

* **onda-packages:** add onda packages ([b619064](https://github.com/vtex/onda/commit/b619064d04d190d5615a2832dbcb86d96efd16ee))





# [0.1.0-labs.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.2...@vtex/onda-react@0.1.0-labs.3) (2021-05-19)


### Features

* **onda:** allow global styles from theme ([aff3fa4](https://github.com/vtex/onda/commit/aff3fa449c6b92004312bf8ee4f648d7e6258dc0))
* **onda-core:** transform theme values on css variables ([3a8a379](https://github.com/vtex/onda/commit/3a8a379c3ccebf4738b65d39fc38aa8ac594e621))
* **onda-runtime-emotion:** allow default entries ([73a0dea](https://github.com/vtex/onda/commit/73a0deafda0b18ca09ca6eca00637d0c976055d8))





# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
