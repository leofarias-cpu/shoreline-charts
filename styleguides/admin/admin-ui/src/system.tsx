import React, { ReactNode } from 'react'
import {
  createSystem,
  useTheme,
  useClassName,
  createElement,
  useResponsiveValue,
  get,
  merge,
  StyleProp,
  StyleObject,
} from '@vtex/admin-ui-system'
import { css, Global } from '@emotion/react'
import { theme } from '@vtex/admin-ui-theme'
import 'focus-visible/dist/focus-visible'

interface ThemeProviderProps {
  children?: ReactNode
}

const { ThemeProvider: BaseProvider, cn, stylesOf } = createSystem(theme)

function ThemeProvider(props: ThemeProviderProps) {
  const { children } = props

  return (
    <BaseProvider>
    <Global
      styles={css`
            @import 'https://io.vtex.com.br/fonts/vtex-trust/style.css';

            code > * > *,
            pre > * > *,
            code,
            pre {
              font-family: 'SF Mono', 'Operator Mono', 'Dank Mono',
                'Fira Code Retina', 'Fira Code', 'FiraCode-Retina', Consolas, Monaco,
                monospace !important;
            }

            body {
              margin: 0;
              background-color: ${get(theme, 'colors.background')};
              color: ${get(theme, 'colors.text')};
            }

            html,
            body,
            * {
              font-family: VTEXTrustVF, -apple-system, system-ui, BlinkMacSystemFont,
                sans-serif !important;
              font-variation-settings: 'wght' 92;
            }

            *,
            ::before,
            ::after {
              box-sizing: border-box;
              border-width: 0;
              border-style: solid;
            }

            html,
            body,
            div,
            span,
            applet,
            object,
            iframe,
            h1,
            h2,
            h3,
            h4,
            h5,
            h6,
            p,
            blockquote,
            pre,
            a,
            abbr,
            acronym,
            address,
            big,
            cite,
            code,
            del,
            dfn,
            em,
            img,
            ins,
            kbd,
            q,
            s,
            samp,
            small,
            strike,
            strong,
            sub,
            sup,
            tt,
            var,
            b,
            u,
            i,
            center,
            dl,
            dt,
            dd,
            ol,
            ul,
            li,
            fieldset,
            form,
            label,
            legend,
            table,
            caption,
            tbody,
            tfoot,
            thead,
            tr,
            th,
            td,
            article,
            aside,
            canvas,
            details,
            embed,
            figure,
            figcaption,
            footer,
            header,
            hgroup,
            menu,
            nav,
            output,
            ruby,
            section,
            summary,
            time,
            mark,
            audio,
            video {
              margin: 0;
              padding: 0;
              border: 0;
              font-size: 100%;
              font: inherit;
              vertical-align: baseline;
            }

            /* HTML5 display-role reset for older browsers */
            article,
            aside,
            details,
            figcaption,
            figure,
            footer,
            header,
            hgroup,
            menu,
            nav,
            section {
              display: block;
            }

            body {
              line-height: 1;
            }

            blockquote,
            q {
              quotes: none;
            }

            blockquote:before,
            blockquote:after,
            q:before,
            q:after {
              content: '';
              content: none;
            }

            table {
              border-collapse: collapse;
              border-spacing: 0;
            }
          `}
        />
      {children}
    </BaseProvider>
  )
}

export {
  ThemeProvider,
  cn,
  useClassName,
  useTheme,
  useResponsiveValue,
  get,
  merge,
  theme,
  StyleProp,
  StyleObject,
  stylesOf,
  createElement,
}
