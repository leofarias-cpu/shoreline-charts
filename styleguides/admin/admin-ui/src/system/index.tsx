import React, { ReactNode } from 'react'
import {
  createSystem,
  useTheme,
  useClassName,
  jsxs,
  useResponsiveValue,
  get,
  merge,
  StyleProp,
  StyleObject,
  cx,
  injectGlobal,
} from '@vtex/admin-ui-system'
import { CacheProvider } from '@emotion/react'
import { theme } from '@vtex/admin-ui-theme'
import 'focus-visible/dist/focus-visible'
import createCache from '@emotion/cache'
interface ThemeProviderProps {
  children?: ReactNode
  appId: string
}

const { ThemeProvider: BaseProvider, cn, stylesOf } = createSystem(theme)

function ThemeProvider(props: ThemeProviderProps) {
  const { children, appId } = props

  return (
    <BaseProvider>
      <CacheProvider value={createCache({ key: `vtex-admin-ui-${appId}` })}>
        {injectGlobal`
            @import 'https://io.vtex.com.br/fonts/vtex-trust/style.css';

            body {
              margin: 0;
              background-color: ${get(theme, 'colors.light.primary')};
              color: ${get(theme, 'colors.dark.primary')};
            }

            html,
            body {
              font-family: ${get(theme, 'fonts.sans')} !important;
              font-variation-settings: ${get(theme, 'fontSettings.regular')};
            }

            strong,
            b {
              font-family: ${get(theme, 'fonts.sans')} !important;
              font-variation-settings: ${get(theme, 'fontSettings.bold')};
            }

            pre,
            code {
              font-family: ${get(theme, 'fonts.mono')} !important;
            }

            * {
              font-family: ${get(theme, 'fonts.sans')} !important;
              font-variation-settings: ${get(theme, 'fontSettings.regular')};
            }

            *,
            ::before,
            ::after {
              box-sizing: border-box;
              border-width: 0;
              border-style: solid;
            }

            html,
            body,
            div,
            span,
            applet,
            object,
            iframe,
            h1,
            h2,
            h3,
            h4,
            h5,
            h6,
            p,
            blockquote,
            a,
            abbr,
            acronym,
            address,
            big,
            cite,
            del,
            dfn,
            em,
            img,
            ins,
            kbd,
            q,
            s,
            samp,
            small,
            strike,
            strong,
            sub,
            sup,
            tt,
            var,
            b,
            u,
            i,
            center,
            dl,
            dt,
            dd,
            ol,
            ul,
            li,
            fieldset,
            form,
            label,
            legend,
            table,
            caption,
            tbody,
            tfoot,
            thead,
            tr,
            th,
            td,
            article,
            aside,
            canvas,
            details,
            embed,
            figure,
            figcaption,
            footer,
            header,
            hgroup,
            menu,
            nav,
            output,
            ruby,
            section,
            summary,
            time,
            mark,
            audio,
            video {
              margin: 0;
              padding: 0;
              border: 0;
              font-size: 100%;
              font: inherit;
              vertical-align: baseline;
            }

            /* HTML5 display-role reset for older browsers */
            article,
            aside,
            details,
            figcaption,
            figure,
            footer,
            header,
            hgroup,
            menu,
            nav,
            section {
              display: block;
            }

            body {
              line-height: 1;
            }

            blockquote,
            q {
              quotes: none;
            }

            blockquote:before,
            blockquote:after,
            q:before,
            q:after {
              content: '';
              content: none;
            }

            table {
              border-collapse: collapse;
              border-spacing: 0;
            }
          `}
        {children}
      </CacheProvider>
    </BaseProvider>
  )
}

export {
  ThemeProvider,
  cn,
  useClassName,
  useTheme,
  useResponsiveValue,
  get,
  merge,
  theme,
  StyleProp,
  StyleObject,
  stylesOf,
  jsxs,
  cx,
}
