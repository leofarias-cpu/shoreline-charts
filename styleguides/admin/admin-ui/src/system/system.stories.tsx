import React from 'react'
import { Meta, Story } from '@storybook/react'

import { useSystem } from './index'

export default {
  title: 'system',
} as Meta

export const cxAndCn: Story = () => {
  const { cn, cx } = useSystem()

  return (
    <div
      className={cx('my-className', cn({ bg: 'blue', color: 'light.primary' }))}
    >
      Cn
    </div>
  )
}
