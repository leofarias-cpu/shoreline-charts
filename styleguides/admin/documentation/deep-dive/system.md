---
path: /docs/deep-dive/system
---

# Admin UI System

> üêã This is a deep dive content. Is recommended for new developers within admin-ui or enthusiasts.

```bash
yarn add @vtex/admin-ui-system
---
npm install @vtex/admin-ui-system
```

`react` and `react-dom` are peer dependencies, so make sure that you also installed them.

## createElement

The `createElement` function calls `React.createElement` using a passed reakit component as base.

```jsx static
import { Box } from 'reakit/Box'
import { createElement } from '@vtex/admin-ui-system'

type Props = {
  /** */
}

function NewComponent(props: Props) {
  return createElement({
    component: Box,
    htmlProps: props,
  })
}
```

It also suports forward refs.

```jsx static
import { forwardRef, Ref } from 'react'
import { Box } from 'reakit/Box'
import { createElement } from '@vtex/admin-ui-system'

type Props = {
  /** */
}

function NewComponent(props: Props, ref: Ref<HTMLDivElement>) {
  return createElement({
    component: Box,
    htmlProps: props,
    ref: ref,
  })
}

const NewComponent = forwardRef(_NewComponent)
```

## cleanProps

It returns the corret htmlProps without strange props.

```js static
const props = {
  children: 'some text',
  id: 'id',
  sx: { c: 'blue' }
  styles: { bg: 'black' },
  palette: 'primary',
}

console.log(cleanProps(props)) // { children: 'some text', id: 'id' }
```

## useClassName

```jsx static
import { Box } from 'reakit/Box'
import { createElement, useClassName } from '@vtex/admin-ui-system'

type Props = {
  /** */
}

function NewComponent(props: Props) {
  const cn = useClassName({ props })
  return createElement({
    component: Box,
    htmlProps: props,
  })
}
```

## className
