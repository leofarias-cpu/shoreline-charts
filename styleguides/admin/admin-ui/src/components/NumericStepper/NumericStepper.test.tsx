import React from 'react'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'

import { NumericStepper } from './index'
import { ThemeProvider } from '../../system'

describe('NumericStepper tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <NumericStepper
          value={1}
          onChange={() => {}}
          label="numeric-stepper"
          data-testid="numeric-stepper"
          styleOverrides={{ bg: 'azure' }}
        />
      </ThemeProvider>
    )

    expect(getByTestId('numeric-stepper')).toHaveStyleRule(
      'background-color',
      'azure'
    )
  })

  it('should match snapshot', async () => {
    const { container } = render(
      <ThemeProvider>
        <NumericStepper value={1} onChange={() => {}} label="stepper number" />
        <NumericStepper
          value={1}
          onChange={() => {}}
          helperText="Helper Text"
          label="numeric-stepper-1"
        />
        <NumericStepper
          value={9}
          onChange={() => {}}
          disabled
          label="numeric-stepper-2"
        />
        <NumericStepper
          value={0}
          error
          onChange={() => {}}
          errorMessage="Minimum of 6"
          label="numeric-stepper-3"
        />
        <NumericStepper
          value={0}
          minValue={0}
          maxValue={50}
          unitMultiplier={5}
          onChange={(event) => {
            console.warn(event.value)
          }}
          label="numeric-stepper-4"
        />
      </ThemeProvider>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })

  it('should not have any violations', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <NumericStepper
          value={1}
          onChange={() => {}}
          maxValue={3}
          minValue={-3}
          helperText="Helper Text"
          label="numeric-stepper"
        />
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
