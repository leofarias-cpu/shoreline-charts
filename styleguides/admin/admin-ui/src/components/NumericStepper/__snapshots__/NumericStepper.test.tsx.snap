// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericStepper tests should not have any violations 1`] = `
<DocumentFragment>
  .emotion-9 {
  line-height: 1.25;
  font-variation-settings: 'wght' 92;
  font-size: 0.875rem;
  width: 106px;
}

.emotion-7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 48px;
  width: 106px;
  padding-top: 12px;
  padding-bottom: 12px;
  line-height: 1.25;
  font-variation-settings: 'wght' 92;
  font-size: 0.875rem;
}

.emotion-3 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: absolute;
  color: #2953B2;
  background-color: transparent;
  font-size: 0.75rem;
  height: 40px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  left: 0.25rem;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-3:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-3:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-3:disabled {
  color: #898F9E;
}

.emotion-3 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-2 {
  line-height: 1.25;
  font-variation-settings: 'wght' 92;
  font-size: 0.875rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
}

.emotion-6 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: absolute;
  color: #2953B2;
  background-color: transparent;
  font-size: 0.75rem;
  height: 40px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  right: 0.25rem;
}

.emotion-6:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-6:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-6:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-6:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-6:disabled {
  color: #898F9E;
}

.emotion-6 svg {
  width: 20px;
  min-width: 20px;
  height: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-0 {
  width: 106px;
  height: 48px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-left: 36px;
  padding-right: 36px;
  text-align: center;
  color: #323845;
  border-style: solid;
  border-width: 1px;
  border-color: #898F9E;
  border-radius: 4px;
  background-color: #FFFFFF;
}

.emotion-0::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0rem;
}

.emotion-0:hover {
  border-color: #323845;
}

.emotion-0:disabled {
  background-color: #F4F6FB;
  color: #898F9E;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-8 {
  line-height: 1.125;
  font-variation-settings: 'wght' 92;
  font-size: 0.75rem;
  color: #898F9E;
  margin-top: 0.5rem;
}

<div
    class="emotion-9"
  >
    <div
      class="emotion-7"
    >
      <input
        aria-label="numeric-stepper"
        class="emotion-0"
        max="3"
        min="-3"
        type="number"
        value="1"
      />
      <button
        aria-label="numeric-stepper-decrease-button"
        class="emotion-3"
        type="button"
      >
        <div
          class="emotion-2"
        >
          <svg
            class="emotion-1"
            fill="none"
            viewBox="0 0 24 24"
          >
            <path
              clip-rule="evenodd"
              d="M18.4112 12C18.4112 12.4142 18.0755 12.75 17.6612 12.75H6.34754C5.93333 12.75 5.59754 12.4142 5.59754 12C5.59754 11.5858 5.93333 11.25 6.34754 11.25H17.6612C18.0755 11.25 18.4112 11.5858 18.4112 12Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <button
        aria-label="numeric-stepper-increase-button}"
        class="emotion-6"
        type="button"
      >
        <div
          class="emotion-2"
        >
          <svg
            class="emotion-1"
            fill="none"
            viewBox="0 0 24 24"
          >
            <path
              clip-rule="evenodd"
              d="M12.0045 5.59557C12.4187 5.59557 12.7545 5.93136 12.7545 6.34557L12.7545 17.6593C12.7545 18.0735 12.4187 18.4093 12.0045 18.4093C11.5903 18.4093 11.2545 18.0735 11.2545 17.6593L11.2545 6.34557C11.2545 5.93136 11.5903 5.59557 12.0045 5.59557Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M18.4114 12.0024C18.4114 12.4166 18.0756 12.7524 17.6614 12.7524H6.34766C5.93345 12.7524 5.59766 12.4166 5.59766 12.0024C5.59766 11.5882 5.93345 11.2524 6.34766 11.2524H17.6614C18.0756 11.2524 18.4114 11.5882 18.4114 12.0024Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="emotion-8"
    >
      Helper Text
    </div>
  </div>
</DocumentFragment>
`;
